woocommerce/compatibility/class-sv-wc-order-compatibility.php

```diff
@@ -522,6 +522,121 @@ public static function get_edit_order_url( \WC_Order $order ) {
+     *
+	 * @return mixed
+	 */
+	public static function get_order_meta( int $order_id, string $key, $single = null) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+		$value = false;
+
+		// setting $single to false by default if hpos is disabled to match it with get_post_meta method's default value
+		if( is_null($single) ) {
+			$single = $is_hpos_enabled;
+		}
+
+		if ( $is_hpos_enabled ) {
+			$order = wc_get_order($order_id);
+			if( $order ){
+				$value = $order->get_meta($key, $single);
+			}
+		} else {
+			$value = get_post_meta( $order_id, $key, $single );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Updates the order meta according to HPOS availability.
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional update_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+     * @param mixed $value Required. Meta value.
+     *
+	 */
+	public static function update_order_meta( int $order_id, string $key, $value ) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+
+		if ( $is_hpos_enabled ) {
+			$order = wc_get_order($order_id);
+			if( $order ){
+				$order->update_meta_data($key, $value, $order_id);
+				$order->save();
+			}
+		} else {
+			update_post_meta( $order_id, $key, $value );
+		}
+	}
+
+
+	/**
+	 * Adds the order meta according to HPOS availability.
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional add_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+     * @param mixed $value Required. meta value.
+	 * @param bool $unique Optional. Whether the same key should not be added.
+	 * 			   Default false.
+     *
+	 */
+	public static function add_order_meta( int $order_id, string $key, $value, $unique = false ) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+
+		if ( $is_hpos_enabled ) {
+			$order = wc_get_order($order_id);
+			if( $order ){
+				$order->add_meta_data( $key, $value, $unique );
+				$order->save();
+			}
+		} else {
+			add_post_meta( $order_id, $key, $value, $unique );
+		}
+	}
+
+	/**
+	 * Deletes the order meta according to HPOS availability.
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional delete_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+	 * @param mixed $meta_value Optional. Applicable only if the HPOS is inactive.
+     *
+	 */
+	public static function delete_order_meta( int $order_id, string $key ) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+
+		if ( $is_hpos_enabled ) {
+			$order = wc_get_order($order_id);
+			if( $order ){
+				$order->delete_meta_data( $key );
+				$order->save();
```

Comment:
no need for saving the whole order object

Reply:
follow up discussion: https://github.com/godaddy-wordpress/wc-plugin-framework/pull/581#discussion_r1065926128